lte/gateway/c/session_manager/LocalSessionManagerHandler.cpp

```diff
@@ -392,12 +334,112 @@ void LocalSessionManagerHandlerImpl::send_create_session(
+        auto err_status =
+            Status(grpc::ABORTED, "Failed to update SessionStore");
+        send_local_create_session_response(err_status, sid, cb);
+        return;
+      }
+      MLOG(MINFO) << "Successfully recycled an existing CWF session " << sid;
+      send_local_create_session_response(grpc::Status::OK, sid, cb);
+      return;
+    }
+  }
+  MLOG(MINFO) << "Found a non-active CWF session with the same IMSI " << imsi
+              << ", requesting a new session";
+  send_create_session(session_map, sid, cfg, cb);
+}
+
+void LocalSessionManagerHandlerImpl::handle_create_session_lte(
+    SessionMap& session_map, const LocalCreateSessionRequest& request,
+    const std::string& sid, SessionConfig cfg,
+    std::function<void(Status, LocalCreateSessionResponse)> cb) {
+  auto imsi       = request.sid().id();
+
+  // If there are no existing sessions for the IMSI, just create a new one
+  auto it = session_map.find(imsi);
+  if (it == session_map.end()) {
+    send_create_session(session_map, sid, cfg, cb);
+    return;
+  }
+
+  for (const auto& session : it->second) {
+    if (cfg == session->get_config() && session->is_active()) {
+      // To recycle the session, it has to be active (i.e., not in
+      // transition for termination) AND it should have the exact same
+      // configuration.
+      MLOG(MINFO) << "Found an active completely duplicated session with IMSI "
+                  << imsi << " and APN " << cfg.apn << ", and same "
+                  << "configuration. Recycling the existing session " << sid;
+      send_local_create_session_response(grpc::Status::OK, sid, cb);
+      return;  // Return early
+    }
+    // At this point, we have session with same IMSI, but not identical config
+    if (session->get_config().apn == cfg.apn && session->is_active()) {
+      // If we have found an active session with the same IMSI+APN, but NOT
+      // identical context, we should terminate the existing session.
+      MLOG(MINFO) << "Found an active session with the same IMSI " << imsi
+                  << " and APN " << cfg.apn << ", but different "
+                  << "configuration. Ending the existing session, "
+                  << "and requesting a new session";
+      LocalEndSessionRequest end_session_req;
+      end_session_req.mutable_sid()->CopyFrom(request.sid());
+      end_session_req.set_apn(cfg.apn);
+      end_session(
+          session_map, end_session_req,
+          [&](grpc::Status status, LocalEndSessionResponse response) {});
+    }
+  }
+  send_create_session(session_map, sid, cfg, cb);
+}
+
+void LocalSessionManagerHandlerImpl::send_local_create_session_response(
+    Status status, const std::string& sid,
+    std::function<void(Status, LocalCreateSessionResponse)> response_callback) {
+  try {
+    LocalCreateSessionResponse resp;
+    resp.set_session_id(sid);
+    response_callback(status, resp);
+  } catch (...) {
```

Comment:
is that try catch new? If so, why you added it here?

Reply:
It's not new, I just moved this block into a function because it was used in a few different places